### PR TITLE
chore!: migrate project resource & datasource

### PR DESCRIPTION
**Pull request by HuyPhanNguyen**
Migration for project resource and data source
[sc-84548](https://app.shortcut.com/octopusdeploy/story/84548/projects-data-source)
[sc-84172](https://app.shortcut.com/octopusdeploy/story/84172/project-resource)

### CODE REVIEW

File: octopusdeploy_framework/resource_project.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+func (r *projectResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan projectResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	project := expandProject(ctx, plan)
+	// Password doesn't return from API so this is work around
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/resource_project.go:45]: Not sure what this comment is referring to...is it still necessary?

File: octopusdeploy_framework/resource_project.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+			gitRef = "main"
+		}
+	}
+
+	deploymentSettings, err := r.Client.Deployments.GetDeploymentSettings(project, gitRef)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading deployment settings: %w", err))
+	}
+
+	// Update the state with the deployment settings
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/resource_project.go:205]: Looks like this should be a few lines up

File: octopusdeploy_framework/resource_project_flatten.go [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+			if snowSettings, ok := extensionSetting.(*projects.ServiceNowExtensionSettings); ok {
+				model.ServiceNowExtensionSettings = flattenServiceNowExtensionSettings(snowSettings)
+			}
+		}
+	}
+
+	return model, diags
+}
+
+func processPersistenceSettings(ctx context.Context, project *projects.Project, model *projectResourceModel, diags diag.Diagnostics) (diag.Diagnostics, *projectResourceModel, diag.Diagnostics, bool) {
```
Comment (hnrkndrssn) [inline octopusdeploy_framework/resource_project_flatten.go:101]: There's two diagnostics return values, is that intentional?

Comment (hnrkndrssn) [reply octopusdeploy_framework/resource_project_flatten.go:101]: And what is the boolean supposed to be about?

**Review by hnrkndrssn — CHANGES_REQUESTED** [review]
A question about the return values from one of the new functions, and some comments about comments

File: octopusdeploy_framework/schemas/project.go [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+					Attributes: map[string]resourceSchema.Attribute{
+						"connection_id":                       util.ResourceString().Required().Description("The connection identifier associated with the extension settings.").Build(),
+						"is_enabled":                          util.ResourceBool().Required().Description("Specifies whether or not this extension is enabled for this project.").Build(),
+						"is_state_automatically_transitioned": util.ResourceBool().Required().Description("Specifies whether or not this extension will automatically transition the state of a deployment for this project.").Build(),
+						"standard_change_template_name":       util.ResourceString().Optional().Description("The name of the standard change template associated with this extension. If provided, deployments will create a standard change based on the provided template, otherwise a normal change will be created.").Build(),
+					},
+				},
+				Description: "Provides extension settings for the ServiceNow integration for this project.",
+			},
+			"template": resourceSchema.ListNestedBlock{
```
Comment (benPearce1) [inline octopusdeploy_framework/schemas/project.go:118]: Got some errors when setting up templates:
```
template {
    name = "template"
  }
```

>│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to octopusdeploy_project.project, provider "provider[\"octopus.com/com/octopusdeploy\"]" produced an unexpected new value: .template[0].label: was
│ null, but now cty.StringVal("").
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to octopusdeploy_project.project, provider "provider[\"octopus.com/com/octopusdeploy\"]" produced an unexpected new value:
│ .template[0].display_settings: was null, but now cty.MapValEmpty(cty.String).
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.

**Review by benPearce1 — APPROVED** [review]
Some issues discovered with variable template state, other than it looks good.

**Pull request merged**